gslab_scons/release.py

```diff
@@ -97,10 +45,61 @@
     local_release = '/Users/%s/Google Drive/release/%s/' % (USER, name)
     local_release = local_release + version + '/'
     
-    release(vers              = version, 
-            DriveReleaseFiles = release_files,
-            local_release     = local_release, 
-            org               = organisation, 
-            repo              = repo,
-            target_commitish  = branch,
-            zip_release       = zip_release)
+    rtools.release(vers              = version, 
+                   DriveReleaseFiles = release_files,
+                   local_release     = local_release, 
+                   org               = organisation, 
+                   repo              = repo,
+                   target_commitish  = branch,
+                   zip_release       = zip_release)
+
+
+def inspect_repo():
+    '''Ensure the repo is ready for release.'''
+    if not rtools.up_to_date(mode = 'scons'):
+        raise ReleaseError('SCons targets not up to date.')
+    elif not rtools.up_to_date(mode = 'git'):
+        print "Warning: `scons` has run since your latest git commit.\n"
+        response = raw_input("Would you like to continue anyway? (y|n)\n")
+        if response in ['N', 'n']: 
+            sys.exit()
+
+
+def issue_size_warnings(file_MB_limit, total_MB_limit, bytes_in_MB):
+    '''Issue warnings if the files versioned in release are too large'''
```

Comment:
Everything that is going to be versioned has been versioned, yes. I think the idea is to remind the user that they have probably done something wrong since the last release, which may be better than never issuing size warnings. I agree it would be better, in principle, to perform this check at the commit-stage. When I originally introduced the size warnings, I was mostly following [this comment](https://github.com/gslab-econ/template/issues/19#issuecomment-262024738) and some other notes in its issue.